color.d

```diff
@@ -128,23 +128,64 @@ struct Color {
 	}
 
 @safe:
-	/++
-		The color components are available as a static array, individual bytes, and a uint inside this union.
 
-		Since it is anonymous, you can use the inner members' names directly.
-	+/
-	union {
-		ubyte[4] components; /// [r, g, b, a]
-
-		/// Holder for rgba individual components.
-		struct {
-			ubyte r; /// red
-			ubyte g; /// green
-			ubyte b; /// blue
-			ubyte a; /// alpha. 255 == opaque
+	ubyte[4] components; /// [r, g, b, a]
+
+	alias components this; /// enables `foreach` etc.
```

Comment:
I am a little iffy on this, alias this has a lot of side effects and there's no guarantee any random ubyte[4] i s in the right rgba sequence and this would allow that assignment, If we really want foreach I think opApply is a better choice.... but i kinda say just foreach(part; color.components) is plenty good enough... that reads a b it more clearly too.

Reply:
Given that it’s fairly easy to just use `components` instead and there isn’t much value to the side effects, I’d say let’s revert the *alias this*